appservice/src/tree/AppSettingsTreeItem.ts

```diff
@@ -10,12 +10,19 @@ import { AppSettingTreeItem } from './AppSettingTreeItem';
 import { getThemedIconPath } from './IconPath';
 import { ISiteTreeRoot } from './ISiteTreeRoot';
 
-export function validateAppSettingKey(settings: StringDictionary, newKey?: string, oldKey?: string): string | undefined {
-    newKey = newKey ? newKey.trim() : '';
-    oldKey = oldKey ? oldKey.trim().toLowerCase() : oldKey;
+export function validateAppSettingKey(settings: StringDictionary, newKey?: string, oldKey?: string, isLinux: boolean = false): string | undefined {
+    // Default to Windows webapp (isLinux = false) to let users access looser name validation, even if saving the key fails
```

Comment:
I'm not sure about the wording here.  How about?

`Default to Windows web app because there are less strict constraints on app setting names for Windows.  If the key is invalid, Azure will prevent adding the key with an error.`

Reply:
Why do y'all need a default? Isn't `isLinux` always defined on SiteClient?